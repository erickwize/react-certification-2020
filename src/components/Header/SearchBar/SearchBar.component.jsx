import React, { useState } from  'react';

import styled from 'styled-components';

const SearchBox = styled.div`
    background-color:grey;
    align-self: center;
    margin-left: 1em;
    height: 2em;
    border: none;
    border-radius: 0.5em;
`;

const Input = styled.input `
    border: none;
    margin-top: 0.5em;
    background-color: grey;
    font-size: 0.9em;
    color: white;
    outline: none;
    .focus{
        color: white;
    }
`;


const SearchBar = () => {
    const [search, setSearch] = useState("Wizeline")

    return (
        <>
            <SearchBox> 
                <img src="https://upload.wikimedia.org/wikipedia/commons/0/0b/Search_Icon.svg" 
                alt="" width="25" height="25" 
                style={{verticalAlign: 'middle'}} />
                <Input 
                placeholder="Search..." 
                type="text" 
                value={search}
                onChange={(evt) => setSearch(evt.target.value)}
                onClick={(evt) => evt.target.select}/>
            </SearchBox>
        </>
    );
};

export default SearchBar;