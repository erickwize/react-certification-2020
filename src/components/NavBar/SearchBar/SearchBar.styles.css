.searchBar{
    display: block;
    position: relative;
    background-color: rgba(255,255,255,0.25);
    min-width: 20%;
    height: 4.5vh;
    border-radius: 8px;
    left: 6px;
}

.searchLogo{
    display: inline-flex;
    justify-content: center;
    height: 4.5vh;
    line-height: 4.5vh;
    width: 30px;
}
.searchIcon{
    display: block;
    position: relative;
    line-height: 30px;
    top: 5px;
    left: 2px;
    color: #f0efef;

}

.searchTxt{
    height: 4.5vh;
    border: none;
    outline: none;
    background: none;
    color: #f0efef;
    line-height: 3.5vh;
}

@media only screen and (max-width: 850px) {
    .searchBar{
        min-width: 45%;
        height: 34px;
    }
}