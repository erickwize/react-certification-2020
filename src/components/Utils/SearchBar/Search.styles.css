.search {
  width: auto;
  margin-left: 24px;
  position: relative;
  margin-right: 16px;
  border-radius: 4px;
  background-color: rgba(255, 255, 255, 0.15);
}

.search__icon {
  fill: currentColor;
  width: 1em;
  height: 1em;
  display: inline-block;
  font-size: 1.5rem;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  flex-shrink: 0;
  user-select: none;
}

.search__icon__container {
  height: 100%;
  display: flex;
  padding: 0px 16px;
  position: absolute;
  align-items: center;
  pointer-events: none;
  justify-content: center;
}

.search__input {
  padding: 8px 8px 8px 0px;
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding-left: calc(1em + 32px);
  font: inherit;
  color: currentColor;
  border: 0;
  height: 1.1876em;
  margin: 0;
  display: block;
  min-width: 0;
  background: none;
  box-sizing: content-box;
  animation-name: mui-auto-fill-cancel;
  letter-spacing: inherit;
  animation-duration: 10ms;
  -webkit-tap-highlight-color: transparent;
}

.search__input__container {
  cursor: text;
  display: inline-flex;
  position: relative;
  font-size: 1rem;
  box-sizing: border-box;
  align-items: center;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  font-weight: 400;
  line-height: 1.1876em;
  letter-spacing: 0.00938em;
}
