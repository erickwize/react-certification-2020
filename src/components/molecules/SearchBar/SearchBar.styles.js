import styled from 'styled-components';
import { VscSearch } from 'react-icons/vsc';

export const SearchContainer = styled.section`
  display: flex;
  flex-direction: row;
  align-items: left;
`;

export const SearchIcon = styled(VscSearch)`
  display: block;
  font-size: 20px;
  color: #6c56a3;
  border: none;
  cursor: not-allowed;
`;

export const Form = styled.form`
  display: flex;
  flex-direction: row;
`;

export const Input = styled.input`
  padding: 0.4rem 0.6rem;
  width: 80%;
  height: 35px;
  margin: 0.2em 0;
  color: #f894f1;
  font-weight: 600;
  font-size: 0.8rem;
  background: white;
  border: 2px solid;
  border-color: #6c56a3;
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
`;

export const Button = styled.button`
  width: 20%;
  height: 35px;
  margin: 0.2em 0;
  margin-left: -0.2em;
  padding: 0.35rem 0.5rem;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  border: 2px solid;
  border-color: #6c56a3;
  align-items: center;
  justify-content: center;
`;

/* background-color: rgba(0, 0, 0, 0.1); */
