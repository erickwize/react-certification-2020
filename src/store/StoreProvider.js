import React from 'react';
import {createContext, useContext, useReducer, useEffect } from 'react';
import storeReducer, { initialStore, types } from './StoreReducer';

const  StoreContext = createContext();

const StoreProvider = ({children}) => {
    const [store, dispatch] = useReducer(storeReducer, initialStore);
    useEffect(() => {
        const data = localStorage.getItem('favorite-videos');
        if(data){
            dispatch({
                type: types.setFromLocalStorage,
                payload: JSON.parse(data)
            })
            dispatch({
                type: types.updateFavorite,
                payload: JSON.parse(data)[ JSON.parse(data).length - 1]
            })
        }
    },[])
    useEffect(() => {
        localStorage.setItem('favorite-videos', JSON.stringify(store.favoriteVideos));
    })
    return(
        <StoreContext.Provider value={[store, dispatch]}>
            {children}
        </StoreContext.Provider>
    )
}

const useStore = () => useContext(StoreContext)[0];
const useDispatch = () => useContext(StoreContext)[1];

export { StoreContext, useStore, useDispatch }; 
export default StoreProvider;