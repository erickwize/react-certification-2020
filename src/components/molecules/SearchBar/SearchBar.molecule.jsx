import React from 'react';
import { Button, Form, Input, SearchContainer, SearchIcon } from './SearchBar.styles';

const SearchBar = () => {
  return (
    <SearchContainer>
      <Form>
        <Input type="text" id="name" placeholder="Ask for a video..." />
        <Button>
          <SearchIcon />
        </Button>
      </Form>
    </SearchContainer>
  );
};

export default SearchBar;
