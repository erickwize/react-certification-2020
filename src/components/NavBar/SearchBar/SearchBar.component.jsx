import React, { useState } from 'react';
import { useDispatch } from '../../../store/StoreProvider';
import { types } from '../../../store/StoreReducer';
import { getVideosByQuery } from '../../../store/dispatchCalls';
import {useHistory} from 'react-router';
import './SearchBar.styles.css';

export default function SearchBar(){
    const history = useHistory();
    const dispatch = useDispatch();
    const [value, setValue] = useState('wizeline');

    const updateValue = e => { setValue(e.target.value)}

    const updateVideos = () => getVideosByQuery(value, dispatch);

    const updateSection = () => dispatch({ type: types.setSectionToMain });

    const onSubmitHandle = e =>{
        e.preventDefault();
        history.push('/');
        updateVideos();
        updateSection();
    }

    return(
        <form onSubmit={onSubmitHandle} className="searchBar">
            <span className="searchLogo">
                <i className="fas fa-search searchIcon"></i>
            </span>
            <input className="searchTxt" type="text" placeholder="Search" value={value} onChange={updateValue}/>
        </form>
    );
}