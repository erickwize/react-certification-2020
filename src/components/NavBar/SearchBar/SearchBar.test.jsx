import React from 'react';
import {render, screen} from '@testing-library/react';
import SearchBar from './SearchBar.component';
import StoreProvider from '../../../store/StoreProvider';

describe('Search Bar', () => {
    it('renders', () => {
        render(
            <StoreProvider>
                <SearchBar/>
            </StoreProvider>
        )
        const searchBar = screen.getByPlaceholderText("Search")
        expect(searchBar).toBeInTheDocument();
    });

    it('starts with wizeline value', () => {
        render(
            <StoreProvider>
                <SearchBar/>
            </StoreProvider>
        )
        const searchBar = screen.getByPlaceholderText("Search")
        expect(searchBar.value).toBe("wizeline");
    });

    it('matches snapshot', () => {
        render(
            <StoreProvider>
                <SearchBar/>
            </StoreProvider>
        )
        const searchBar = screen.getByPlaceholderText("Search")
        expect(searchBar).toMatchSnapshot();
    });
})