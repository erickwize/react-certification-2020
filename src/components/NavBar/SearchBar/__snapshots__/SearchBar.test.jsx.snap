// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar matches snapshot 1`] = `
<input
  class="searchTxt"
  placeholder="Search"
  type="text"
  value="wizeline"
/>
`;
